### PR TITLE
Change to build-type: Simple

### PR DESCRIPTION
**Pull request by phadej**
`cabal new-build` generates `.ghc.environment` files, so we
can run `doctest` on sources, and package db info will be picked implicitly.
This works on GHC >= 8.0. In short

   cabal new-install -z doctest
   cabal new-build
   doctest src

If there aren't `-z` flag in your `cabal`, use a workaround to install
doctest:

   (cd /tmp && cabal new-install doctest)

`stack` users can `stack exec doctest src` too, as `stack exec` will
provide package environment in an environment variable.

The only downside, is that `doctest` binary have to correspond
to the GHC in use. This is not a problem in CI, but is inconvenient
locally.

### CODE REVIEW

**Comment by phadej** [comment]
Green, "resolves" #13

File: .travis.yml [hunk 1]
```diff
@@ -119,6 +116,9 @@ install:
       cabal new-update head.hackage -v
     fi
   - grep -Ev -- '^\s*--' ${HOME}/.cabal/config | grep -Ev '^\s*$'
+  - (cd /tmp && echo '' | cabal new-repl -w ${HC} --build-dep fail)
+  - if [ $HCNUMVER -ge 80000 ]; then cabal new-install -w ${HC} -j2 --symlink-bindir=$HOME/.local/bin doctest --constraint='doctest ==0.16.*'; fi
```
Comment (RyanGlScott) [inline .travis.yml]: > `if [ $HCNUMVER -ge 80000 ]`

I'm rather uncomfortable with hard-coding the GHC version to an old release like this.

File: .travis.yml [hunk 2]
```diff
@@ -119,6 +116,9 @@ install:
...
+  - if [ $HCNUMVER -eq 80403 ]; then cabal new-install -w ${HC} -j2 --symlink-bindir=$HOME/.local/bin hlint --constraint='hlint ==2.1.*'; fi
```
Comment (RyanGlScott) [inline .travis.yml]: Similarly here.

Comment (phadej) [reply .travis.yml]: There's an issue about this in `haskell-ci` and I'll look into it (no promises when though).

**Comment by phadej** [comment]
FWIW, this is a demo. It's up to Neil to persuade you Ryan. No offence taken if this PR is rejected.

File: .travis.yml (hunk 1 above)
Comment (RyanGlScott) [reply .travis.yml]: To be clear, I'm not suggesting that we run this on every version of GHC, but only on the latest release.

File: .travis.yml (hunk 2 above)
Comment (RyanGlScott) [reply .travis.yml]: Unfortunately, I'm not willing to accept this PR unless this issue is resolved.

If we do decide to go with this option, I'll add a hack to `travis-maintenance` for this.

File: .travis.yml (hunk 1 above)
Comment (phadej) [reply .travis.yml]: it's `-ge` so it will run on all versions `>= 8.0` (currently doctests are run for all versions)

**Comment by phadej** [comment]
Does all the travis-maintenance features have a corresponding ticket in haskell-ci?

**Comment by RyanGlScott** [comment]
[Here](https://github.com/RyanGlScott/travis-maintenance/blob/f3a84aa35cbf68f274be5d3c42a68a5500c1c254/src/Hacks.hs#L19-L23) are all of the hacks that I currently use in `travis-maintenance`:

```haskell
data Hack
  = HLint [String]
  | DisableTestsGlobally
  | AlternateConfig [String]
  | CabalProjectMiscellanea [String]
```

Here is (in my mind) what each hack maps to in terms of `haskell-ci` issues:

1. `HLint`: haskell-CI/haskell-ci#176
2. `DisableTestsGlobally`: I couldn't find one specifically for this issue, but I only use this to work around ku-fpg/blank-canvas#73, and I'm reasonably sure that there's a better way to accomplish this (that I haven't figured out yet).
3. `AlternateConfig`: haskell-CI/haskell-ci#182
4. `CabalProjectMiscellanea`: haskell-CI/haskell-ci#184

File: .travis.yml (hunk 1 above)
Comment (RyanGlScott) [reply .travis.yml]: Ah, that makes this slightly more tolerable, then. I can live this slightly suboptimal state of affairs.

**Pull request merged**